### PR TITLE
feat(Text): Deprecate Text / Heading and add more robust UI Text Component [GM-162]

### PR DESCRIPTION
**Pull request by codecaaron**
#Overview

Monolith PR: https://github.com/codecademy-engineering/Codecademy/pull/23373
Static Sites PR: https://github.com/codecademy-engineering/static-sites/pull/703

## Deprecations
`Text`=> `TextDeprecated`
`Heading` => `HeadingDeprecated`

Migration Strategy:
1. Cutover all current uses of these components to `Deprecated` versions.
2. Symmetric Updates: Remove current `Text` from gamut-labs and port to new version.  Removing unneeded props on the way. 
3. Asymmetric Migrations: Update `Heading` and `Text` separately to new usage with a map to the correct fontSizes and removing unneeded custom props.
4. Fully Remove them once done.

#### S Y M M E T R Y 
tumblr_p72gqfd2Kg1vymbt8o4_400

### New Props
- Typography props: `whiteSpace`, `textTransform`
- Position props: `opacity`

### Moved Props
- `justifySelf` and `alignSelf` are now on Layout to be more distributed and not tied to flex.

### `<Text />`

New `<Text />` component with more sensible defaults per each `as` possibility.  This component covers both heading and text while also giving overridable default values to  each variant. 

This should drastically reduce the amount of configuration + overrides needed.

```tsx
variants: {
    h1: {
      fontSize: 64,
      fontWeight: 'title',
      lineHeight: 'title',
    },
    h2: {
      fontSize: 44,
      fontWeight: 'title',
      lineHeight: 'title',
    },
    h3: {
      fontSize: 34,
      fontWeight: 'title',
      lineHeight: 'title',
    },
    h4: {
      fontSize: 26,
      fontWeight: 'title',
      lineHeight: 'title',
    },
    h5: {
      fontSize: 22,
      fontWeight: 'title',
      lineHeight: 'title',
    },
    h6: {
      fontSize: 20,
      fontWeight: 'title',
      lineHeight: 'title',
    },
    small: {
      fontSize: 14,
      display: 'inline-block',
    },
    p: {
      fontSize: 16,
    },
    strong: {
      fontWeight: 'title',
    },
    code: {
      fontFamily: 'monospace',
    },
    span: {
      display: 'inline-block',
    },
  },
```

* Deprecates old CSS Module text components in favor of a new flexible `Text` component built for UI text.

### PR Checklist

- [ ] Related to designs:
- [x] Related to JIRA ticket: GM-162
- [x] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.2.40-alpha.63cb22.0<br/>@codecademy/gamut-labs@13.2.3-alpha.63cb22.0<br/>@codecademy/gamut@29.4.7-alpha.63cb22.0<br/>@codecademy/styleguide@33.3.8-alpha.63cb22.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://606152d200036987270c0f1e--gamut-preview.netlify.app>https://606152d200036987270c0f1e--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/606152d200036987270c0f1e>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6061e40291a4d22cfe2f1d13--gamut-preview.netlify.app>https://6061e40291a4d22cfe2f1d13--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6061e40291a4d22cfe2f1d13>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.2.40-alpha.40c126.0<br/>@codecademy/gamut-labs@13.2.3-alpha.40c126.0<br/>@codecademy/gamut@29.4.7-alpha.40c126.0<br/>@codecademy/styleguide@33.3.8-alpha.40c126.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6061f29e2a897c408ed22692--gamut-preview.netlify.app>https://6061f29e2a897c408ed22692--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6061f29e2a897c408ed22692>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.9.5-alpha.6e6de3.0<br/>@codecademy/gamut-kit@0.2.40-alpha.6e6de3.0<br/>@codecademy/gamut-labs@13.2.3-alpha.6e6de3.0<br/>@codecademy/gamut-styles@9.1.2-alpha.6e6de3.0<br/>@codecademy/gamut-system@0.5.3-alpha.6e6de3.0<br/>@codecademy/gamut-tests@2.3.2-alpha.6e6de3.0<br/>@codecademy/gamut@29.4.7-alpha.6e6de3.0<br/>@codecademy/konamimojisplosion@0.2.2-alpha.6e6de3.0<br/>@codecademy/markdown-overrides@0.5.15-alpha.6e6de3.0<br/>@codecademy/styleguide@33.3.8-alpha.6e6de3.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60623c9dd74c9300a461dde0--gamut-preview.netlify.app>https://60623c9dd74c9300a461dde0--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60623c9dd74c9300a461dde0>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.9.5-alpha.9c89d6.0<br/>@codecademy/gamut-kit@0.2.40-alpha.9c89d6.0<br/>@codecademy/gamut-labs@13.2.3-alpha.9c89d6.0<br/>@codecademy/gamut-styles@9.1.2-alpha.9c89d6.0<br/>@codecademy/gamut-system@0.5.3-alpha.9c89d6.0<br/>@codecademy/gamut-tests@2.3.2-alpha.9c89d6.0<br/>@codecademy/gamut@29.4.7-alpha.9c89d6.0<br/>@codecademy/konamimojisplosion@0.2.2-alpha.9c89d6.0<br/>@codecademy/markdown-overrides@0.5.15-alpha.9c89d6.0<br/>@codecademy/styleguide@33.3.8-alpha.9c89d6.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://606244badd3fe11a508e5160--gamut-preview.netlify.app>https://606244badd3fe11a508e5160--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/606244badd3fe11a508e5160>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.9.5-alpha.4277f1.0<br/>@codecademy/gamut-kit@0.2.41-alpha.4277f1.0<br/>@codecademy/gamut-labs@13.2.4-alpha.4277f1.0<br/>@codecademy/gamut-styles@9.1.2-alpha.4277f1.0<br/>@codecademy/gamut-system@0.5.3-alpha.4277f1.0<br/>@codecademy/gamut-tests@2.3.2-alpha.4277f1.0<br/>@codecademy/gamut@29.4.8-alpha.4277f1.0<br/>@codecademy/konamimojisplosion@0.2.2-alpha.4277f1.0<br/>@codecademy/markdown-overrides@0.5.15-alpha.4277f1.0<br/>@codecademy/styleguide@33.3.9-alpha.4277f1.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6062471982e65e1be72cd64c--gamut-preview.netlify.app>https://6062471982e65e1be72cd64c--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6062471982e65e1be72cd64c>Deploy Logs</a></p>

File: packages/gamut/__tests__/__snapshots__/gamut-test.ts.snap [hunk 1]
```diff
@@ -82,6 +82,8 @@ Array [
   "Text",
   "TextArea",
   "TextButton",
+  "TextDeprecated",
+  "textProps",
```
Comment (dreamwasp) [inline packages/gamut/__tests__/__snapshots__/gamut-test.ts.snap:86]: why do we need to export textProps like this?

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.9.5-alpha.9eff3f.0<br/>@codecademy/gamut-kit@0.2.41-alpha.9eff3f.0<br/>@codecademy/gamut-labs@13.2.4-alpha.9eff3f.0<br/>@codecademy/gamut-styles@9.1.2-alpha.9eff3f.0<br/>@codecademy/gamut-system@0.5.3-alpha.9eff3f.0<br/>@codecademy/gamut-tests@2.3.2-alpha.9eff3f.0<br/>@codecademy/gamut@29.4.8-alpha.9eff3f.0<br/>@codecademy/konamimojisplosion@0.2.2-alpha.9eff3f.0<br/>@codecademy/markdown-overrides@0.5.15-alpha.9eff3f.0<br/>@codecademy/styleguide@33.3.9-alpha.9eff3f.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://606252bc9c42c4349c663e65--gamut-preview.netlify.app>https://606252bc9c42c4349c663e65--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/606252bc9c42c4349c663e65>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.9.5-alpha.6f52a1.0<br/>@codecademy/gamut-kit@0.2.41-alpha.6f52a1.0<br/>@codecademy/gamut-labs@13.2.4-alpha.6f52a1.0<br/>@codecademy/gamut-styles@9.1.2-alpha.6f52a1.0<br/>@codecademy/gamut-system@0.5.3-alpha.6f52a1.0<br/>@codecademy/gamut-tests@2.3.2-alpha.6f52a1.0<br/>@codecademy/gamut@29.4.8-alpha.6f52a1.0<br/>@codecademy/konamimojisplosion@0.2.2-alpha.6f52a1.0<br/>@codecademy/markdown-overrides@0.5.15-alpha.6f52a1.0<br/>@codecademy/styleguide@33.3.9-alpha.6f52a1.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60625485550178303c501b7e--gamut-preview.netlify.app>https://60625485550178303c501b7e--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60625485550178303c501b7e>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.9.5-alpha.b1da2f.0<br/>@codecademy/gamut-kit@0.2.41-alpha.b1da2f.0<br/>@codecademy/gamut-labs@13.2.4-alpha.b1da2f.0<br/>@codecademy/gamut-styles@9.1.2-alpha.b1da2f.0<br/>@codecademy/gamut-system@0.5.3-alpha.b1da2f.0<br/>@codecademy/gamut-tests@2.3.2-alpha.b1da2f.0<br/>@codecademy/gamut@29.4.8-alpha.b1da2f.0<br/>@codecademy/konamimojisplosion@0.2.2-alpha.b1da2f.0<br/>@codecademy/markdown-overrides@0.5.15-alpha.b1da2f.0<br/>@codecademy/styleguide@33.3.9-alpha.b1da2f.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://606257356f26563683cbc18a--gamut-preview.netlify.app>https://606257356f26563683cbc18a--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/606257356f26563683cbc18a>Deploy Logs</a></p>

**Review by dreamwasp — COMMENTED** [review]
lol ignore me

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.9.5-alpha.a1f625.0<br/>@codecademy/gamut-kit@0.2.41-alpha.a1f625.0<br/>@codecademy/gamut-labs@13.2.4-alpha.a1f625.0<br/>@codecademy/gamut-styles@9.1.2-alpha.a1f625.0<br/>@codecademy/gamut-system@0.5.3-alpha.a1f625.0<br/>@codecademy/gamut-tests@2.3.2-alpha.a1f625.0<br/>@codecademy/gamut@29.4.8-alpha.a1f625.0<br/>@codecademy/konamimojisplosion@0.2.2-alpha.a1f625.0<br/>@codecademy/markdown-overrides@0.5.15-alpha.a1f625.0<br/>@codecademy/styleguide@33.3.9-alpha.a1f625.0<br/></pre>

**Review by dreamwasp — APPROVED** [review]
live it love it 🚢 it

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60625df0c9cfa34dad0813b3--gamut-preview.netlify.app>https://60625df0c9cfa34dad0813b3--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60625df0c9cfa34dad0813b3>Deploy Logs</a></p>

File: packages/gamut/__tests__/__snapshots__/gamut-test.ts.snap (hunk 1 above)
Comment (codecaaron) [reply packages/gamut/__tests__/__snapshots__/gamut-test.ts.snap]: We don't :D

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.9.5-alpha.29677c.0<br/>@codecademy/gamut-kit@0.2.41-alpha.29677c.0<br/>@codecademy/gamut-labs@13.2.4-alpha.29677c.0<br/>@codecademy/gamut-styles@9.1.2-alpha.29677c.0<br/>@codecademy/gamut-system@0.5.3-alpha.29677c.0<br/>@codecademy/gamut-tests@2.3.2-alpha.29677c.0<br/>@codecademy/gamut@29.4.8-alpha.29677c.0<br/>@codecademy/konamimojisplosion@0.2.2-alpha.29677c.0<br/>@codecademy/markdown-overrides@0.5.15-alpha.29677c.0<br/>@codecademy/styleguide@33.3.9-alpha.29677c.0<br/>@codecademy/variance@0.1.2-alpha.29677c.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60627a4b27be9f07eea5887a--gamut-preview.netlify.app>https://60627a4b27be9f07eea5887a--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60627a4b27be9f07eea5887a>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.9.6-alpha.d9be1a.0<br/>@codecademy/gamut-kit@0.2.42-alpha.d9be1a.0<br/>@codecademy/gamut-labs@13.2.5-alpha.d9be1a.0<br/>@codecademy/gamut-styles@9.1.3-alpha.d9be1a.0<br/>@codecademy/gamut-system@0.5.3-alpha.d9be1a.0<br/>@codecademy/gamut-tests@2.3.3-alpha.d9be1a.0<br/>@codecademy/gamut@29.4.9-alpha.d9be1a.0<br/>@codecademy/konamimojisplosion@0.2.3-alpha.d9be1a.0<br/>@codecademy/markdown-overrides@0.5.16-alpha.d9be1a.0<br/>@codecademy/styleguide@33.3.10-alpha.d9be1a.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60627cee471f3a1d6bc1def7--gamut-preview.netlify.app>https://60627cee471f3a1d6bc1def7--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60627cee471f3a1d6bc1def7>Deploy Logs</a></p>

**Pull request merged**